dev/com.ibm.ws.security.wim.adapter.ldap/src/com/ibm/ws/security/wim/adapter/ldap/BEROutputStream.java

```diff
@@ -0,0 +1,94 @@
+/*******************************************************************************
+ * Copyright (c) 2020 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.security.wim.adapter.ldap;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+
+import com.ibm.websphere.ras.Tr;
+import com.ibm.websphere.ras.TraceComponent;
+import com.ibm.websphere.ras.annotation.Trivial;
+
+/******************************************************************************
+*	Use an extended version of ByteArrayOutputStream in order to allow access
+*	to protected values.
+******************************************************************************/
+@Trivial
+//rename BER output stream
+public class BEROutputStream extends ByteArrayOutputStream {
+    private static final TraceComponent tc = Tr.register(BEROutputStream.class);
+    public final String eol = getSystemProperty("line.separator");
+    @Override
+    public void write(int b) {
+        String METHODNAME = "write(int b)";
+        super.write(b);
+        try {
+            if (this.toString("UTF-8").contains(eol)) {
+                try {
+                    super.writeTo(System.out);
+                    super.reset();
+                } catch (IOException e) {
+                    if (tc.isDebugEnabled())
+                        Tr.debug(tc, METHODNAME + " Cannot write to System out: " + e.toString());
+                }
+            }
+        } catch (UnsupportedEncodingException e) {
+            if (tc.isDebugEnabled())
+                Tr.debug(tc, METHODNAME + " UTF-8 is not a valid encoding: " + e.toString());
+        }
+    }
+
+    @Override
+    public void write(byte[] b, int off, int len) {
+        String METHODNAME = "write(byte[] b, int off, int len)";
+        super.write(b, off, len);
+        try {
+            if (this.toString("UTF-8").contains(eol)) {
```

Comment:
```suggestion
            if (this.toString("UTF-8").contains(System.lineSeparator())) {
```

Reply:
I also wonder if the BER output uses different line separators on windows vs unix? Have you tested on both?